flink-runtime/src/main/java/org/apache/flink/runtime/io/network/buffer/BufferConsumer.java

```diff
@@ -115,7 +131,8 @@ public Buffer build() {
 	 * @return a retained copy of self with separate indexes
 	 */
 	public BufferConsumer copy() {
-		return new BufferConsumer(buffer.retainBuffer(), writerPosition.positionMarker, currentReaderPosition);
+		checkState(isShareable, "The underlying buffer is not shareable.");
```

Comment:
From this point, my previous concern of whether the `isShareable` tag is indicating for both buffer & event and only for buffer might have the answer. The semantic should be covering both buffer and event, otherwise the event copy would destroy this check, although the event shareable would not be used in practice atm.

Reply:
We might supplement some descriptions for this method to warn that only shareable `BufferConsumer` can support `copy` function.